Merge master into develop and other changes for X-Plane 12 compatibility

NPrincen opened the pull request:
Repository master is 7 commits ahead of develop.  Merge the recent changes to master into develop, so it has all the latest changes.

Introduced a corrected version of "Updated DataRefs for Xplane 12 #288" to fix Issue #286.

federeghe committed:
[Build] CameraCallbacks.cpp is missing in CMakeLists.txt

jason-watkins committed:
Merge pull request #244 from federeghe/master

[Build] CameraCallbacks.cpp is missing in CMakeLists.txt

Commit:
Making the Python 3 package pip-installable

emanuelen5 committed:
Apply suggestions from code review

Co-authored-by: Christopher Teubert <christopher.a.teubert@nasa.gov>

emanuelen5 committed:
Unifying quotation marks and removing comment

emanuelen5 committed:
Making the version same as the XPC plugin

jason-watkins committed:
Merge pull request #265 from emanuelen5/master

Making the Python 3 package pip-installable

NPrincen committed:
Merge branch 'master' into develop

NPrincen committed:
Update DataManager.cpp

Resolves Issue #286 and fixes the code from the pull request #288.

jason-watkins commented:
Unfortunately our policy says that merging to `master` requires a formal release, which includes a whole lot of paperwork I don't have time for.

NPrincen commented:
Sorry, but I think you misunderstood the merge.  I was asking for the 7 commits that are in master that are not in develop to be merged into develop.  Right now, master has changes that are not in develop.  I think you can do this without doing the paperwork for a formal release.

Also, I went and marked this Merge Request as Draft right now.  I am working to make XPlaneConnect X-Plane 12 compatible.  It doesn't work because of a few issues.  There are errors being generated that need to be fixed.  getTERR seems to be throwing an error with the previous plugin.  I would like to update to the latest 4.01 SDK and see if that fixes it.  So, I am not ready for you to merge changes to develop yet because I am still working on it.

Maybe you want me to split this into two Merges?  One for the port of master changes to develop and the other for the X-Plane 12 changes.

Thank You,
Norm Princen
Boeing

jason-watkins commented:
Fair enough. I did indeed misunderstand. As long as the diff isn't too big, I'm fine with accepting 1 PR for both.

NPrincen committed:
Update SDK to 4.01

Update the SDK to the latest X-Plane SDK 4.0.1.  Keep the old 32 bit libraries from X-Plane SDK 2.1.3.

NPrincen committed:
Update Drawing.cpp

Changed xplm_Phase_Objects to xplm_Phase_Modern3D to be compatible with XPLM302 and newer.

NPrincen committed:
Update MessageHandlers.cpp

Deleted call to XPLMCommandKeyStroke because it has been deprecated.  Replace later.

NPrincen committed:
Update CMakeLists.txt

Added definitions for all XPLM200 through XPLM400 to be compatible with the 4.01 SDK and X-Plane 12.

NPrincen committed:
Update xpcPlugin.vcxproj

Added definitions for all XPLM200 through XPLM400 to be compatible with the 4.01 SDK and X-Plane 12.

NPrincen committed:
Update xpcPlugin.vcxproj

Added definitions for all XPLM200 through XPLM400 to be compatible with the 4.01 SDK and X-Plane 12.

NPrincen committed:
Update project.pbxproj

Added definitions for all XPLM200 through XPLM400 to be compatible with the 4.01 SDK and X-Plane 12.

teddybear082 commented:
Hi OP just want to confirm - does your fork fully work with Xplane 12? I am trying to create a project for xplane 12 with python so if so I will just use your fork since your changes haven’t been merged.  Thanks!
